### ISSUE TITLE
Ellipses and Rectangle Shapes don't check for invalid imputs

### ISSUE THREAD

**Issue by Carreau** [Napari/napari#4289]
Ellipses and rectangle do not check that the "4 corners" representations are actually 4 corners, 
and are not crossed (for rectangles). 

I started t look into it in #4273 but as the default for shapes is rectangles, and almost all test passes random, non rectangles-shaped 4 corners data to `Shape()`, so for now I gave up updating. 

I'm also not even sure we want to prevent invalid data. Do we ?

Both also say

```
data : (4, D) array ...
        ....
        ...., or a (4, D) array specifying the four corners of a
        bounding box that contains the ellipse. These need not be axis aligned.
```

I don't think D != 2 is tested.

**Comment by psobolewskiPhD** [comment]
Indeed, not only do crossed rectangle work, any 4 vertex appear to work, e.g.
```
data = np.asarray([[0, 0], [0, 5], [5, 10], [5, 5]])
```
